### PR TITLE
Issue 822  ray trafo impl default

### PR DESCRIPTION
**Pull request by adler-j**
I'm keeping with the theme of the day of removing code, i.m.o. this is much cleaner. An issue is that users are not given sufficient information that they *REALLY* want to install ASTRA.

### CODE REVIEW

File: odl/tomo/operators/ray_trafo.py [hunk 1]
```diff
@@ -57,18 +57,19 @@ def __init__(self, discr_domain, geometry, impl='astra_cpu', **kwargs):
...
+        ----------------
+        impl : {`None`, 'astra_cuda', 'astra_cpu', 'scikit'}, optional
             Implementation back-end for the transform. Supported back-ends:
-            'astra_cpu': ASTRA toolbox using CPU, only 2D
-            'astra_cuda': ASTRA toolbox, using CUDA, 2D or 3D
-            'scikit': scikit-image, only 2D parallel with square domain
+            * 'astra_cuda': ASTRA toolbox, using CUDA, 2D or 3D
+            * 'astra_cpu': ASTRA toolbox using CPU, only 2D
+            * 'scikit': scikit-image, only 2D parallel with square domain
+            If `None` is given, prefers the fastest available backend.
```
Comment (kohr-h) [inline odl/tomo/operators/ray_trafo.py]: \`\`None\`\`
the fastest available back-end is used

File: odl/tomo/operators/ray_trafo.py [hunk 2]
```diff
@@ -246,17 +260,18 @@ def __init__(self, discr_range, geometry, impl='astra_cpu', **kwargs):
...
+        Other Parameters
+        ----------------
         impl : {'astra_cpu', 'astra_cuda', 'scikit'}, optional
             Implementation back-end for the transform. Supported back-ends:
-            'astra_cpu': ASTRA toolbox using CPU, only 2D
-            'astra_cuda': ASTRA toolbox, using CUDA, 2D or 3D
-            'scikit': scikit-image, only 2D parallel with square domain
+            * 'astra_cuda': ASTRA toolbox, using CUDA, 2D or 3D
+            * 'astra_cpu': ASTRA toolbox using CPU, only 2D
+            * 'scikit': scikit-image, only 2D parallel with square domain
```
Comment (kohr-h) [inline odl/tomo/operators/ray_trafo.py]: put strings in double backticks

**Comment by adler-j** [comment]
> Small stuff, easy to fix

Any comments on ASTRA? Should we write a few more words in the readme perhaps?

**Comment by adler-j** [comment]
Rebased from master, waiting for CI.

**Comment by kohr-h** [comment]
> Any comments on ASTRA? Should we write a few more words in the readme perhaps?

Hm, perhaps something like "By default, the fastest available back-end is chosen." wouldn't hurt.

**Comment by adler-j** [comment]
> Hm, perhaps something like "By default, the fastest available back-end is chosen." wouldn't hurt.
Not sure if it makes sense to put it in the front page readme.

Simply feels weird to have comments like that everywhere, yet it will probably be the first question people have "ODL is slow, what do?" Perhaps we leave it as is now and think a bit about it?

**Comment by kohr-h** [comment]
> Simply feels weird to have comments like that everywhere, yet it will probably be the first question people have "ODL is slow, what do?" Perhaps we leave it as is now and think a bit about it?

Fine with me, go ahead.

**Comment by kohr-h** [comment]
Also, users that don't do anything non-standard won't have a single back-end installed by default, so they rather ask "Tomography don't work, what do?".

**Pull request merged**